erpnext/regional/report/uae_vat_201/uae_vat_201.py

```diff
@@ -0,0 +1,336 @@
+	"""Append emiratewise standard rated expenses and vat."""
+	for d, emirate in enumerate(emirates, 97):
+		if emirate in amounts_by_emirate:
+			amounts_by_emirate[emirate]["no"] = _('1{0}').format(chr(d))
+			amounts_by_emirate[emirate]["legend"] = _('Standard rated supplies in {0}').format(emirate)
+			data.append(amounts_by_emirate[emirate])
+		else:
+			append_data(data, _('1{0}').format(chr(d)),
+				_('Standard rated supplies in {0}').format(emirate),
+				frappe.format(0, 'Currency'), frappe.format(0, 'Currency'))
+	return amounts_by_emirate
+
+def append_vat_on_expenses(data, filters):
+	"""Appends Expenses and All Other Inputs."""
+	append_data(data, '', _('VAT on Expenses and All Other Inputs'), '', '')
+	append_data(data, '9', _('Standard Rated Expenses'),
+		frappe.format(get_standard_rated_expenses_total(filters), 'Currency'),
+		frappe.format(get_standard_rated_expenses_tax(filters), 'Currency'))
+
+	append_data(data, '10', _('Supplies subject to the reverse charge provision'),
+		frappe.format(get_reverse_charge_recoverable_total(filters), 'Currency'),
+		frappe.format(get_reverse_charge_recoverable_tax(filters), 'Currency')
+)
+
+def append_data(data, no, legend, amount, vat_amount):
+	"""Returns data with appended value."""
+	data.append({"no": no, "legend":legend, "amount": amount, "vat_amount": vat_amount})
+
+def get_total_emiratewise(filters):
+	"""Returns Emiratewise Amount and Taxes."""
+	query_filters = get_filters(filters)
+	query_filters['docstatus'] = ['=', 1]
+	return frappe.db.get_list('Sales Invoice', 
+		filters = query_filters,
+		fields = ['vat_emirate as emirate','sum(total)', 'sum(total_taxes_and_charges)'],
+		group_by='vat_emirate',
+		as_list=True
+	)
+
+def get_emirates():
+	"""Returns a List of emirates in the order that they are to be displayed."""
+	return [
+		'Abu Dhabi',
+		'Dubai',
+		'Sharjah',
+		'Ajman',
+		'Umm Al Quwain',
+		'Ras Al Khaimah',
+		'Fujairah'
+	]
+
+def get_filters(filters):
+	"""The conditions to be used to filter data to calculate the total sale."""
+	query_filters = {}
+	if filters.get("company"):
+		query_filters["company"] = ['=', filters['company']]
+	if filters.get("from_date"):
+		query_filters["posting_date"] = ['>=', filters['from_date']]
+	if filters.get("from_date"):
+		query_filters["posting_date"] = ['<=', filters['to_date']]
+	return query_filters
+
+def get_reverse_charge_total(filters):
+	"""Returns the sum of the total of each Purchase invoice made."""
+	query_filters = get_filters(filters)
+	query_filters['reverse_charge'] = ['=', 'Y']
+	query_filters['docstatus'] = ['=', 1]
+	return frappe.db.get_list('Purchase Invoice', 
+		filters = query_filters,
+		fields = ['sum(total)'],
+		as_list=True,
+		limit = 1
+	)[0][0]  or 0
+
+def get_reverse_charge_tax(filters):
+	"""Returns the sum of the tax of each Purchase invoice made."""
+	conditions = get_conditions_join(filters)
+	return frappe.db.sql("""
```

Comment:
You can format the query better, to be much readable.

Reply:
done